app/models/lookup_keys/puppetclass_lookup_key.rb

```diff
@@ -9,6 +9,7 @@ class PuppetclassLookupKey < LookupKey
   after_validation :reset_override_params, :if => ->(key) { key.override_changed? && !key.override? }
 
   scoped_search :relation => :param_classes, :on => :name, :rename => :puppetclass, :aliases => [:puppetclass_name], :complete_value => true
+  scoped_search :relation => :environments, :on => :name, :rename => :environment, :complete_value => true
```

Comment:
:+1: on keeping the `:environments` `:only_explicit`. For the `param_classes`, it would mean a change in a behaviour, that is not related directly to the attached issue. I would probably keep it as separate issue, ideally opened by some user that would hit it.